### PR TITLE
Introduce 2.13

### PR DESCRIPTION
**Pull request by daddykotex**
* drop support for 2.10
* introduce sbt-1.3.0-RC2

Related: https://github.com/dmurvihill/courier/issues/48

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -57,24 +57,39 @@ lazy val commonSettings = releaseSettings ++ Seq(
...
+        "-language:implicitConversions",
+        "-Ypartial-unification"
       )
-    case Some((2, n)) if n == 11 =>
+    case Some((2, n)) if n == 12 =>
       Seq(
         "-language:existentials",
         "-language:higherKinds",
         "-language:implicitConversions",
+        "-Xfuture",
```
Comment (dmurvihill) [inline build.sbt]: It looks like we are removing certain compiler options from our 2.12 build (e.g. `-Ywarn-extra-implicit`, which appears in `ScalacOptions.All` but not in this list, will no longer be used in 2.12).

I noticed Fabian's pull request omitted the same flags. Am I reading it incorrectly?

File: src/main/scala-2.11/compat.scala [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package courier
+
+object Compat {
+  def asJava[T](set: Set[T]): java.util.Set[T] = {
+    import collection.JavaConverters._
+    set.asJava
+  }
+}
```
Comment (dmurvihill) [inline src/main/scala-2.11/compat.scala]: You might need to tweak an editor setting to make sure the ending newline makes it in.

**Comment by daddykotex** [comment]
> Any reason we can't leave 2.10 support in? I'm all for dropping 2.10 if it makes something easier for us but otherwise I'd rather not make a breaking change

get rid of a maintenance burden (keep the build working for 3 versions), compatibility in source code, etc.

> Why SBT 1.3.0-RC2 as opposed to 1.3.3 or 1.4.0?

I'm not aware of those versions, to me, the most recent sbt version is 1.3.0-RC2

File: build.sbt (hunk 1 above)
Comment (daddykotex) [reply build.sbt]: I fixed it. The intent was to remove options from the 2.13 build, because some options are now deprecated.

**Comment by daddykotex** [comment]
I thought about the 2.10 support and I still think that now is a great time to drop it. Publishing to Sonatype got better but it still a pain to publish multiple artifacts. 

Furthermore, it's our only way as a community to move forward. If we keep maintaining old versions, people/compagnies won't see the need to move over. At this point, there should be absolutely no reason one can't move from 2.10 to something more recent.

If you want, you can review that again, I will merge and publish a release candidate by the end of the day if you haven't.

**Pull request merged**